backend/src/test/java/com/google/sps/UserDatastoreTest.java

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.User;
+import com.google.sps.datastore.UserDatastore;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class UserDatastoreTest {
+
+  // Default values
+  private static final long ID_DEFAULT = 0;
+
+  // Some usernames
+  private static final String PERSON_A = "Person A";
+  private static final String PERSON_B = "Person B";
+
+  // Datastore Key/Property constants
+  private static final String KIND_USER = "User";
+  private static final String PROPERTY_USERNAME = "username";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void addOneUser() throws EntityNotFoundException {
+    // Add one user to datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserDatastore userDatastore = new UserDatastore(datastore);
+    User userA = new User(ID_DEFAULT, PERSON_A);
+    userDatastore.addUser(userA);
```

Comment:
tiny nit: this is part of the `act` block of this test and should be separated from the code above and below by blank lines. Same comment on the test right below this one.

Reply:
I'm not sure I understand what you mean. What's the code above line 65?

Reply:
Every unit test can be broken down into three distinct parts: arrange, act, and assert. Each of these blocks should be grouped together with no blank lines within them and with a blank line separating each of them. So two total blank lines per test. 

```
    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
    UserDatastore userDatastore = new UserDatastore(datastore);
    User userA = new User(ID_DEFAULT, PERSON_A);
```

This is the arrange block since it sets up the DatastoreService, UserDatastore, and User objects for the test.

```
 userDatastore.addUser(userA);
```

Is the act block since this test is testing the behavior of addUser. 

```
    Key keyA = KeyFactory.createKey(KIND_USER, PERSON_A);
    Entity entityA = datastore.get(keyA);
    assertThat((String) entityA.getProperty(PROPERTY_USERNAME)).isEqualTo(PERSON_A);
```

This is the assert block since it's verifying that the call in act behaved as expected.

Reply:
I see, thanks for the explanation. Done.